pkg/controller/memberstatus/memberstatus_controller.go

```diff
@@ -200,11 +205,101 @@ func (r *ReconcileMemberStatus) memberOperatorHandleStatus(reqLogger logr.Logger
 	return err
 }
 
+// loadCurrentResourceUsage loads the current usage of the cluster and stores it into the member status
+func (r *ReconcileMemberStatus) loadCurrentResourceUsage(reqLogger logr.Logger, memberStatus *toolchainv1alpha1.MemberStatus) error {
+	memberStatus.Status.ResourceUsage.MemoryUsagePerNodeRole = map[string]int{}
+	allocatableValues, err := r.getAllocatableValues(reqLogger)
+	if err != nil {
+		return err
+	}
+
+	nodeMetricsList := &v1beta1.NodeMetricsList{}
+	if err := r.client.List(context.TODO(), nodeMetricsList); err != nil {
+		return err
+	}
+
+	usagePerRole := map[string]float32{}
+	allocatablePerRole := map[string]float32{}
+NodeMetricsList:
+	for _, nodeMetric := range nodeMetricsList.Items {
+		for key, usage := range nodeMetric.Usage {
```

Comment:
can't you just do `if usage, exists := nodeMetric.Usage["memory"]; exists { ... }` instead?

Reply:
good point :man_facepalming: 
done in 53280b6
thanks for it :+1: